ocs_ci/utility/ibmcloud.py

```diff
@@ -1242,6 +1242,43 @@ def find_free_network_subnets(subnet_cidr, network_prefix=27):
             return possible_subnets
 
 
+def delete_dns_records(cluster_name):
+    """
+    Delete DNS records leftover from cluster destroy.
+
+    Args:
+        cluster_name (str): Name of the cluster, used to filter DNS records
+
+    """
+    dns_domain_id = constants.IBMCLOUD_BASE_DOMAIN_ID
+    cis_instance_name = constants.IBMCLOUD_CIS_INSTANCE_NAME
+    ids_to_delete = []
+    page = 1
+    while True:
+        out = run_ibmcloud_cmd(
+            f"ibmcloud cis dns-records {dns_domain_id} --per-page 1000 --page {page} --output json"
+        )
+        records = json.loads(out)
+        if not records:
+            logger.info("Reached end of pagination")
+            break
+
+        filter_string = f".{cluster_name}."
```

Comment:
Cool, I checked we usually have 3 entries:
```
api.rperiyas-scltest.ibmcloud2.qe.rh-ocs.com
api-int.rperiyas-scltest.ibmcloud2.qe.rh-ocs.com
*.apps.rperiyas-scltest.ibmcloud2.qe.rh-ocs.com
```

So as you are using `.rperiyas-scltest.` it should not by mistake delete other cluster which will have subset of string if person just add just extra number at the end of previous name.

👍

Reply:
That's the exact scenario I wanted to avoid here, since I do that a lot :)